samza-kv/src/test/scala/org/apache/samza/storage/kv/TestLargeMessageSafeStore.scala

```diff
@@ -0,0 +1,187 @@
+
+    val key1 = new Array[Byte](16)
+    val smallMessage = new Array[Byte](32)
+    largeMessageSafeKeyValueStore.put(key1, smallMessage)
+    verify(store, times(1)).put(Matchers.eq(key1), Matchers.eq(smallMessage))
+
+    val key2 = new Array[Byte](16)
+    val largeMessage = new Array[Byte](maxMessageSize + 1)
+    try {
+      largeMessageSafeKeyValueStore.put(key2, largeMessage)
+      fail("The test case should have failed due to a large message being passed to the changelog, but it didn't.")
+    } catch {
+      case e: SamzaException =>
+        verify(store, times(0)).put(Matchers.eq(key2), Matchers.eq(largeMessage))
+    }
+  }
+
+  @Test
+  def testSmallMessagePutAllSuccessWithDropLargeMessageDisabled = {
+    val largeMessageSafeKeyValueStore = new LargeMessageSafeStore(store, storeName, false, maxMessageSize)
+    val key = new Array[Byte](16)
+    val smallMessage = new Array[Byte](32)
+
+    val entries = new ArrayList[Entry[Array[Byte], Array[Byte]]]
+    entries.add(new Entry(key, smallMessage))
+
+    largeMessageSafeKeyValueStore.putAll(entries)
+    verify(store, times(1)).putAll(Matchers.eq(entries))
+    verify(store, times(0)).put(Matchers.any(classOf[Array[Byte]]), Matchers.any(classOf[Array[Byte]]))
+  }
+
+  @Test
+  def testLargeMessagePutAllFailureWithDropLargeMessageDisabled = {
+    val largeMessageSafeKeyValueStore = new LargeMessageSafeStore(store, storeName, false, maxMessageSize)
+    val key = new Array[Byte](16)
+    val largeMessage = new Array[Byte](maxMessageSize + 1)
+
+    val entries = new ArrayList[Entry[Array[Byte], Array[Byte]]]
+    entries.add(new Entry(key, largeMessage))
+
+    try {
+      largeMessageSafeKeyValueStore.putAll(entries)
+      fail("The test case should have failed due to a large message being passed to the changelog, but it didn't.")
+    } catch {
+      case e: SamzaException =>
+        verify(store, times(0)).putAll(Matchers.eq(entries))
+        verify(store, times(0)).put(Matchers.any(classOf[Array[Byte]]), Matchers.any(classOf[Array[Byte]]))
+    }
+  }
+
+  @Test
+  def testSmallMessagePutWithSerdeAndDropLargeMessageDisabled(): Unit = {
+    val largeMessageSafeKeyValueStore = new LargeMessageSafeStore(store, storeName, false, maxMessageSize)
+
+    val longSerde = new LongSerde
+    val longObj = 1000L
+    val key = longSerde.toBytes(longObj)
+
+    val jsonSerde = new JsonSerdeV2[java.util.HashMap[String, Object]]
+    val obj = new java.util.HashMap[String, Object]();
+    obj.put("jack", "jill")
+    obj.put("john", new java.lang.Integer(2))
+    val smallMessage = jsonSerde.toBytes(obj)
+
+    val entries = new ArrayList[Entry[Array[Byte], Array[Byte]]]
+    entries.add(new Entry(key, smallMessage))
+
+    largeMessageSafeKeyValueStore.putAll(entries)
+    verify(store, times(1)).putAll(Matchers.eq(entries))
+    verify(store, times(0)).put(Matchers.any(classOf[Array[Byte]]), Matchers.any(classOf[Array[Byte]]))
```

Comment:
Instead of `verify(store, times(0))`, you can do `verify(store, never())`
Same for other cases below

Reply:
Done.